runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,29 @@
+local M = {}
+
+-- This function checks the first 15 lines for appearance of 'FoamFile'
+-- and then 'object' in a following line.
+-- In that case, it's probably an OpenFOAM file
+function M.foam(path, bufnr)
+  local foam_file = false
+  for _, line in ipairs(vim.api.nvim_get_lines(bufnr, 0, 15, true)) do
+    if line:find("^FoamFile") then
+      foam_file = true
+    elseif foam_file and line:find("^%s*object") then
+      return "foam"
+    end
+  end
+end
+
+-- Determine if a *.tf file is TF mud client or terraform
+function M.tf(path, bufnr)
+  local number_of_lines = vim.fn.line("$")
+  for line = 1, number_of_lines do
+    local current_line = vim.fn.trim(vim.fn.getline(line))
```

Comment:
Would be good to replace this with a local convenience function that takes a `bufnr`(!) and uses `vim.api.nvim_buf_line_count(bufnr)` instead of `getline`. (`trim` is probably also replaceable by a pure Lua function; maybe something like

```lua
local function trim(s)
   local from = s:match"^%s*()"
   return from > #s and "" or s:match(".*%S", from)
end
```

Reply:
Note that we have `vim.trim` as a Lua function already.

Reply:
Oh, nice, I did think that was too useful not to have as a standard library function ;)